File: src/main/java/com/jetbrains/idear/actions/recognition/InlineActionRecognizer.java
package com.jetbrains.idear.actions.recognition;

import com.intellij.openapi.actionSystem.DataContext;
import org.jetbrains.annotations.NotNull;

public class InlineActionRecognizer implements ActionRecognizer {

    @Override
    public boolean isMatching(@NotNull String sentence) {
        return sentence.contains("inline");
    }

    @Override
    public ActionCallInfo getActionInfo(@NotNull String sentence, DataContext dataContext) {
        if (!isMatching(sentence)) return null;
        return new ActionCallInfo("Inline");
    }
}

File: src/main/java/org/openasr/idear/actions/recognition/InlineActionRecognizer.kt
package org.openasr.idear.actions.recognition

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.impl.EditorComponentImpl
import org.openasr.idear.nlp.NlpGrammar
import java.awt.Component

//class InlineActionRecognizer : ActionRecognizer("Inline ") {
//    private val grammar = listOf(
//            NlpGrammar("Inline")
//                    .withExample("inline")
//    )
//
//    override fun isSupported(dataContext: DataContext, component: Component?) = component is EditorComponentImpl
//
//    override fun getGrammars(): List<NlpGrammar> = grammar
//}
